api/operator/v1alpha1/meta.go

```diff
@@ -20,15 +20,15 @@ type ConfigMapReference struct {
 	// name of the ConfigMap.
 	// +kubebuilder:validation:MinLength:=1
 	// +kubebuilder:validation:MaxLength:=253
-	// +kubebuilder:validation:XValidation:rule="!format.dns1123Subdomain().validate(self).hasValue()",message="name must consist of lowercase alphanumeric characters, hyphens ('-'), and periods ('.'). Each block, separated by periods, must start and end with an alphanumeric character. Hyphens are not allowed at the start or end of a block, and consecutive periods are not permitted."
+	// +kubebuilder:validation:Pattern:=`^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$`
 	// +kubebuilder:validation:Required
 	// +required
 	Name string `json:"name"`
 
 	// namespace in which the ConfigMap exists. If empty, ConfigMap will be looked up in IstioCSR created namespace.
 	// +kubebuilder:validation:MinLength:=0
 	// +kubebuilder:validation:MaxLength:=63
-	// +kubebuilder:validation:XValidation:rule=`size(self) == 0 || !format.dns1123Label().validate(self).hasValue()`,message="namespace must consist of only lowercase alphanumeric characters and hyphens, and must start with an alphabetic character and end with an alphanumeric character."
+	// +kubebuilder:validation:Pattern:=`^$|^[a-z0-9]([-a-z0-9]*[a-z0-9])?$`
```

Comment:
`^$` - Matches an empty string:  this field is optional (omitempty)
`^[a-z0-9]([-a-z0-9]*[a-z0-9])?$` - Matches a valid `DNS1123` label